ocaml/xapi-idl/lib/xcp_service.ml

```diff
@@ -476,7 +476,18 @@ let configure ?(options = []) ?(resources = []) () =
 
 let configure2 ~name ~version ~doc ?(options = []) ?(resources = []) () =
   configure_common ~options ~resources @@ fun config_spec ->
-  Cmd.eval ~catch:true (command_of ~name ~version ~doc config_spec)
+  let cmd = command_of ~name ~version ~doc config_spec in
+  match Cmd.eval_value ~catch:true cmd with
+  | Ok (`Ok _) ->
+      ()
+  | Ok `Help | Ok `Version ->
+      exit 0
+  | Error `Parse ->
+      exit 123
+  | Error `Term ->
+      exit 124
+  | Error `Exn ->
+      exit 125
```

Comment:
Instead of hardcoding the numbers you can use the constants from here: https://erratique.ch/software/cmdliner/doc/Cmdliner/Cmd/Exit/index.html